v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/PipelineController.java

```diff
@@ -66,35 +67,14 @@ static PipelineResult executeSingleInstanceMigration(
     Ddl ddl = SpannerSchema.getInformationSchemaAsDdl(spannerConfig);
     ISchemaMapper schemaMapper = PipelineController.getSchemaMapper(options, ddl);
 
-    List<String> tablesToMigrate =
-        PipelineController.listTablesToMigrate(options.getTables(), schemaMapper, ddl);
-    Set<String> tablesToMigrateSet = new HashSet<>(tablesToMigrate);
+    List<String> srcTablesToMigrate = listTablesToMigrate(options.getTables(), schemaMapper, ddl);
```

Comment:
This piece has gotten rather confusing. listTablesToMigrate seems to be doing one set of filtering and then spannerTablesToMigrate another. Can we merge these 2 into 
`listFinalTablesToMigrate(options.getTables(), schemaMapper, ddl)` that can do both sets of filtering in one method ?

Reply:
the list of tablesToMigrate is used in further logic, hence it is not removed here.

To merge the method would require returning a list of lists which would make the code harder to read. Once we simplify the table processing logic, this can be considered.